### PR TITLE
Migrate tests to GitHub Actions

### PR DESCRIPTION
**Pull request by yahavi**
- [x] All [tests](https://ci.appveyor.com/project/jfrog-ecosystem/build-info) passed. If this feature is not already covered by the tests, I added new tests.
-----

Depends on https://github.com/jfrog/build-info/pull/694

Example: https://github.com/yahavi/build-info/actions/runs/3919281662

### CODE REVIEW

File: .github/workflows/integrationTests.yml [hunk 1]
```diff
@@ -0,0 +1,453 @@
...
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+
+      - name: Setup Artifactory
+        run: |
+          go install github.com/jfrog/jfrog-testing-infra/local-rt-setup@latest
+          ~/go/bin/local-rt-setup --rt-version=7.39.4
```
Comment (RobiNino) [inline .github/workflows/integrationTests.yml:293]: Please add a comment explaining why this test is restricted to a specific Artifactory version

File: .github/workflows/integrationTests.yml [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+          RTLIC: ${{secrets.RTLIC}}
+          GOPROXY: direct
+
+      # Run tests
+      - name: Run Tests
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: clean build-info-extractor-npm:test
+        env:
+          BITESTS_PLATFORM_URL: http://localhost:8081
```
Comment (RobiNino) [inline .github/workflows/integrationTests.yml:304]: Please add a comment explaining why the default isn't used

File: build-info-extractor-docker/src/test/java/org/jfrog/build/extractor/docker/extractor/DockerExtractorTest.java [hunk 3]
```diff
@@ -31,44 +27,33 @@
...
     private static final String SHORT_IMAGE_TAG_VIRTUAL = "3";
     private static final String EXPECTED_REMOTE_PATH_KANIKO = "hello-world/latest";
-
-    private static final String LOCAL_DOMAIN = "BITESTS_ARTIFACTORY_DOCKER_LOCAL_DOMAIN";
-    private static final String REMOTE_DOMAIN = "BITESTS_ARTIFACTORY_DOCKER_REMOTE_DOMAIN";
-    private static final String VIRTUAL_DOMAIN = "BITESTS_ARTIFACTORY_DOCKER_VIRTUAL_DOMAIN";
-    private static final String DOCKER_LOCAL_REPO = "BITESTS_ARTIFACTORY_DOCKER_LOCAL_REPO";
-    private static final String DOCKER_REMOTE_REPO = "BITESTS_ARTIFACTORY_DOCKER_REMOTE_REPO";
-    private static final String DOCKER_VIRTUAL_REPO = "BITESTS_ARTIFACTORY_DOCKER_VIRTUAL_REPO";
-    private static final String DOCKER_HOST = "BITESTS_ARTIFACTORY_DOCKER_HOST";
```
Comment (RobiNino) [inline build-info-extractor-docker/src/test/java/org/jfrog/build/extractor/docker/extractor/DockerExtractorTest.java:57]: Doesn't removing the support for host means running on macos is not longer supported?

File: build-info-extractor-npm/src/main/java/org/jfrog/build/extractor/npm/extractor/NpmBuildInfoExtractor.java [hunk 4]
```diff
@@ -38,6 +38,7 @@
 import java.util.Scanner;
 import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
```
Comment (RobiNino) [inline build-info-extractor-npm/src/main/java/org/jfrog/build/extractor/npm/extractor/NpmBuildInfoExtractor.java:41]: Needed?

**Review by RobiNino — CHANGES_REQUESTED** [review]

File: .github/workflows/integrationTests.yml [hunk 5]
```diff
@@ -0,0 +1,453 @@
...
+  # Triggers the workflow on labeled PRs only.
+  pull_request_target:
+    types: [ labeled ]
+# Ensures that only the latest commit is running for each PR at a time.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}-${{ github.ref }}
+  cancel-in-progress: true
+jobs:
+  Distribution:
+    if: contains(github.event.pull_request.labels.*.name, 'safe to test') || github.event_name == 'push'
```
[the remark this answers is not in the document]
Comment (yahavi) [reply .github/workflows/integrationTests.yml:13]: Good idea!

**Pull request merged**